chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -456,17 +469,13 @@ public boolean enterBlock(Scene scene, Ray ray, BlockPalette palette) {
       if (lx != 0 || ly != 0 || lz != 0)
         return false; // outside of octree!
 
-      // Descend the tree to find the current leaf node
-      int level = depth;
-      NodeId node = implementation.getRoot();
-      int nodeIndex = 0;
-      while(implementation.isBranch(node)) {
-        level -= 1;
-        lx = x >>> level;
-        ly = y >>> level;
-        lz = z >>> level;
-        node = implementation.getChild(node, (((lx & 1) << 2) | ((ly & 1) << 1) | (lz & 1)));
-      }
+      Pair<NodeId, Integer> nodeAndLevel = implementation.getWithLevel(x, y, z);
+      NodeId node = nodeAndLevel.getFirst();
+      int level = nodeAndLevel.getSecond();
+
+      lx = x >>> level;
```

Comment:
do we still need the shifts? They were dropped below.

Reply:
Good catch, yes we do, they are used to advance the ray out of the node later. The code below is wrong